sympy/ntheory/generate.py

```diff
@@ -246,9 +263,32 @@ def primepi(n):
     n = int(n)
     if n < 2:
         return 0
-    else:
-        return sieve.search(n)[0]
-
+    lim = int(n ** 0.5)
+    lim -= 1
+    lim = max(lim,0)
+    while lim * lim <= n:
+        lim+=1
+    lim-=1
+    arr1 = [0]*(lim+1)
+    arr2 = [0]*(lim+1)
+    for i in range(1, lim+1):
+        arr1[i] = i-1
```

Comment:
Yeah, phi here and the one in wikipedia are a little different.
All numbers `2<=k<=i` are included in the first step.
I'll write a detailed explanation.